bluebottle/funding_stripe/views.py

```diff
@@ -90,6 +93,123 @@ class StripePaymentList(PaymentList):
     serializer_class = StripePaymentSerializer
 
 
+class StripeBankTransferDetail(JsonApiViewMixin, AutoPrefetchMixin, RetrieveAPIView):
+    queryset = PaymentIntent.objects.all()
+    serializer_class = BankTransferSerializer
+
+    permission_classes = []
+
+    def get_object(self):
+        obj = super().get_object()
+        payment = obj.get_payment()
+        payment.update()
+        obj.refresh_from_db()
+        return obj
+
+
+class StripeBankTransferList(PaymentList):
+    queryset = PaymentIntent.objects.all()
+    serializer_class = BankTransferSerializer
+
+    def perform_create(self, serializer):
+        # Get data from serializer without saving
+        payment_intent_data = serializer.validated_data
+        donation = payment_intent_data['donation']
+        account_currency = str(donation.activity.target.currency)
+        currency = str(donation.amount.currency)
+
+        # Validate the currency compatibility
+        if currency != account_currency:
+            raise ValidationError(f'Bank transfer not supported for currency {currency}')
+
+        # Prepare Stripe and other necessary objects
+        stripe = get_stripe()
+        connect_account = donation.activity.bank_account.connect_account
+        payment_method_options = {}
+
+        # Set up payment method options based on currency
+        if currency == 'EUR':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "eu_bank_transfer",
+                        "eu_bank_transfer": {"country": "NL"},
+                    },
+                },
+            }
+        elif currency == 'USD':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "us_bank_transfer",
+                    },
+                },
+            }
+        elif currency == 'GBP':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "gb_bank_transfer",
+                    },
+                },
+            }
+        elif currency == 'MXN':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "mx_bank_transfer",
+                    },
+                },
+            }
```

Comment:
This looks very duplicated to me. Just calculate the type, and do not repeat the whole objec.t